### ISSUE TITLE
why-depends by name

### ISSUE THREAD

**Issue by lheckemann** [NixOS/nix#6712]
**Is your feature request related to a problem? Please describe.**
I often use `nix why-depends` to determine dependencies between store paths I already have, without evaluation. This is unwieldy though, requiring determining the full store path of what I'm looking for first:

```
$ nix-store -qR /run/current-system | grep openconnect
/nix/store/bh4a67ckvlrangk0mmdy5hqkan8q0lqa-openconnect-8.20

$ nix-why-depends /run/current-system /nix/store/bh4a67ckvlrangk0mmdy5hqkan8q0lqa-openconnect-8.20
/nix/store/704cn9nkmspygcfwdbhbw2gzqy83jb5x-nixos-system-geruest-22.05-20220531-08950a6
└───/nix/store/ix4xvl8gf8z0spig3nqmq8cy8w4s70qz-etc
    └───/nix/store/n834sbzl9d1p0mxak4diyw87vanydcay-user-units
        └───/nix/store/ia7p309sdadja278kcxsfik7i9br0wm9-xdg-desktop-portal-gtk-1.14.0
            └───/nix/store/cm86rqcyx4fy129mpdlmfhi5ir406svr-gnome-settings-daemon-42.1
                └───/nix/store/qg65hz5k9y1zyp9kby16x6pzslgyx5n5-networkmanager-1.38.0
                    └───/nix/store/bh4a67ckvlrangk0mmdy5hqkan8q0lqa-openconnect-8.20
```

**Describe the solution you'd like**
I'd like to be able to run `nix why-depends /run/current-system openconnect` and leave working out which path(s) that means to Magic.

**Describe alternatives you've considered**
I could write a script that wraps Nix, but I'm sure I'm not the only one who wants this, and one solution for everyone would be nice.

**Additional context**
?

**Comment by lheckemann** [comment]
I can, but I want this for everything, not just the special cases where the registry happens to contain the right nixpkgs.

**Comment by LunNova** [comment]
If this goes in it would be good if it also works for `--derivation`. Hopefully that isn't extra work.

Something like this:

```
nix why-depends --derivation .#nixosConfigurations.$(hostname).system.build.toplevel .#nixosConfigurations.$(hostname).pkgs.openssl_3.dev
```

Can guarantee you're using the right `pkgs` instances but it doesn't guarantee the derivation is the right one, maybe it got overridden, maybe it used a different version that you didn't expect.

The use case involving names is also compelling if you want to audit all the versions of a package rather than having to go look for them.

**Comment by nh2** [comment]
Can I use `nix why-depends` with just `.drv`s, without the `#`-flakes way of working?

**Comment by lheckemann** [comment]
Yes, you can give it `.drv`s or built store paths instead of flake refs.
